paperqa/settings.py

```diff
@@ -884,12 +884,18 @@ async def make_ldp_agent(
         raise NotImplementedError(f"Didn't yet handle agent type {agent_type}.")
 
 
-MaybeSettings = Settings | str | None
+# Settings: already Settings
+# dict[str, Any]: serialized Settings
+# str: named Settings
+# None: defaulted Settings
+MaybeSettings = Settings | dict[str, Any] | str | None
 
 
 def get_settings(config_or_name: MaybeSettings = None) -> Settings:
     if isinstance(config_or_name, Settings):
         return config_or_name
+    if isinstance(config_or_name, dict):
+        return Settings(**config_or_name)
```

Comment:
nit but I'm more partial to `.model_validate`

Reply:
Alright moved to it 👌 

Out of curiosity, why do you prefer `model_validate` here? I was playing around, and couldn't actually find a place where `Settings(**config_or_name)` validation fails while `Settings.model_validate(config_or_name)` succeeds